P1684-mdarray/P1684.md

```diff
@@ -334,6 +342,397 @@ In R3 we decided to radically cut back on constructors and simply leave those ou
+                      OtherIndexTypes... exts);
+```
+
+```cpp
+  constexpr mdarray()
+    requires(rank_dynamic() != 0)
+    = default;
+  constexpr mdarray(const mdarray& rhs) = default;
+  constexpr mdarray(mdarray&& rhs) = default;
+
+  template<class... OtherIndexTypes>
+    explicit constexpr mdarray(OtherIndexTypes... exts);
+
+  template<class... OtherIndexTypes>
+    constexpr mdarray(const container_type& c,
+                      OtherIndexTypes... exts);
+
+  template<class... OtherIndexTypes>
+    constexpr mdarray(container_type&& c,
+                      OtherIndexTypes... exts);
+```
+
+---
+
+```cpp
+  explicit constexpr mdarray(const extents_type& ext);
+  explicit constexpr mdarray(const mapping_type& m);
+
+  constexpr mdarray(const container_type& c,
+                    const extents_type& ext);
+  constexpr mdarray(const container_type& c,
+                    const mapping_type& m);
+
+  template<class Alloc>
+    constexpr mdarray(const extents_type& ext,
+                      const Alloc& a);
+  template<class Alloc>
+    constexpr mdarray(const mapping_type& m,
+                      const Alloc& a);
+
+  constexpr mdarray(container_type&& c,
+                    const extents_type& ext);
+  constexpr mdarray(container_type&& c,
+                    const mapping_type& m);
+
+  template<class Alloc>
+    constexpr mdarray(const container_type& c,
+                      const extents_type& ext,
+                      const Alloc& a);
+  template<class Alloc>
+    constexpr mdarray(const container_type& c,
+                      const mapping_type& m,
+                      const Alloc& a);
+
+  template<class Alloc>
+    constexpr mdarray(container_type&& c,
+                      const extents_type& ext,
+                      const Alloc& a);
+  template<class Alloc>
+    constexpr mdarray(container_type&& c,
+                      const mapping_type& m,
+                      const Alloc& a);
+```
+
+```cpp
+  template<class... Args>
+    explicit(sizeof...(Args) == 0)
+    constexpr mdarray(const extents_type& ext,
+                      Args&&... args);
+
+  template<class... Args>
+    explicit(sizeof...(Args) == 0)
+    constexpr mdarray(const mapping_type& m,
+                      Args&&... args);
+```
+
+---
+
+```cpp
+  constexpr mdarray(const extents_type& ext,
+                    const value_type& val);
+  constexpr mdarray(const mapping_type& m,
+                    const value_type& val);
+
+  template<class Alloc>
+    constexpr mdarray(const extents_type& ext,
+                      const value_type& val,
+                      const Alloc& a);
+  template<class Alloc>
+    constexpr mdarray(const mapping_type& m,
+                      const value_type& val,
+                      const Alloc& a);
+```
+
+```cpp
+  template<class... Args>
+    constexpr mdarray(const extents_type& ext,
+                      fill_t,
+                      const value_type& val,
```

Comment:
Should we consider adding an example to clarify why this is needed?  For instance, if `value_type` is `size_t`, just having the `Args&&...` wouldn't be enough to construct the container correctly.